ahds/data_stream.py

```diff
@@ -55,124 +161,59 @@ def byterle_decoder(data, output_size):
                     continue
             elif not count:
                 if repeat:
-                    value = input_data[i:i + no]
+                    #value = input_data[i:i + no]
                     repeat = False
                     count = True
-                    output[j:j+no] = numpy.array(value)
+                    #output[j:j+no] = np.array(value)
+                    output[j:j+no] = input_data[i:i+no]
                     i += no
                     j += no
                     continue
                 elif not repeat:
-                    value = input_data[i]
-                    output[j:j+no] = value
+                    #value = input_data[i]
+                    #output[j:j+no] = value
+                    output[j:j+no] = input_data[i]
                     i += 1
                     j += no
                     count = True
                     repeat = False
                     continue
         
-        assert j == output_size
+        assert j == count
         
         return output
 
+# define common alias for the selected byterle_decoder implementation
 
-def hxbyterle_decode(output_size, data):
-    """Decode HxRLE data stream
-    
-    If C-extension is not compiled it will use a (slower) Python equivalent
+"""Decode HxRLE data stream
+   
+   func:: hxbyterle_decode
+   if C-extension is not compiled it will use a (slower) Python equivalent
     
-    :param int output_size: the number of items when ``data`` is uncompressed
-    :param str data: a raw stream of data to be unpacked
-    :return numpy.array output: an array of ``numpy.uint8``
-    """
-    output = byterle_decoder(data, output_size)
-    assert len(output) == output_size     
-    return output
+   :param int output_size: the number of items when ``data`` is uncompressed
+   :param str data: a raw stream of data to be unpacked
+   :return np.array output: an array of ``np.uint8``
+"""
+hxbyterle_decode = byterle_decoder
 
 
-def hxzip_decode(data_size, data):
+def hxzip_decode(data,dtype=None,count = 0):
     """Decode HxZip data stream
     
     :param int data_size: the number of items when ``data`` is uncompressed
     :param str data: a raw stream of data to be unpacked
-    :return numpy.array output: an array of ``numpy.uint8``
-    """
-    import zlib
-    data_stream = zlib.decompress(data)
-    output = numpy.array(struct.unpack('<{}B'.format(len(data_stream)), data_stream), dtype=numpy.uint8)
-    assert len(output) == data_size
-    return output
-
-
-def unpack_binary(data_pointer, definitions, data):
```

Comment:
Missing.

Reply:
I think i removed it as i think i replaced it by numpy.frombuffer which does the same and has the advantage that it can directly be read in the shape as specified by the file.